pkg/client/factory.go

```diff
@@ -50,9 +54,41 @@ func (a *addQuery) WrappedRoundTripper() http.RoundTripper {
 	return a.next
 }
 
-func NewFactory(cfg *rest.Config, impersonate bool) (*Factory, error) {
+type factoryOptions struct {
+	qps   float32
+	burst int
+}
+
+func defaultFactoryOptions() *factoryOptions {
+	return &factoryOptions{
+		qps:   defaultQPS,
+		burst: defaultBurst,
+	}
+}
+
+// WithQPSAndBurst configures the rest.Config used for creating the clients in
+// the factory with the provided burst and qps configuration.
+//
+// See https://pkg.go.dev/k8s.io/client-go/rest#Config for more.
+func WithQPSAndBurst(qps float32, burst int) FactoryOption {
```

Comment:
With this commit I don't see anything outside test code calling this function. If it's intended to be called from `rancher/rancher/pkg/wrangler/context.go:NewContext` this increases the coupling between the `r/r/pkg/wrangler` code and the steve code.  The change in the first commit takes the config and figures out what to do -- I prefer that to this approach, but can always be convinced.

Reply:
I really don't mind either way.

I'm not sure that it increases the coupling, it was already very tightly coupled (no matter what Rancher passes in as the values via the config, they get overwritten).

This reduces that coupling somewhat, allowing rancher (or any consumer) to configure the values that it wants to see.

But, I'm happy to revert, we want this to allow customers to configure how much we hammer their APIs...so, one way or another we need some sort of change.

Reply:
We'd shift this to Rancher...let it decide what values (likely via the same route).